src/coreclr/vm/tieredcompilation.cpp

```diff
@@ -1041,6 +1128,16 @@ CORJIT_FLAGS TieredCompilationManager::GetJitFlags(PrepareCodeConfig *config)
 
     switch (nativeCodeVersion.GetOptimizationTier())
     {
+        case NativeCodeVersion::OptimizationTierInstrumented:
+            flags.Set(CORJIT_FLAGS::CORJIT_FLAG_BBINSTR);
+            flags.Set(CORJIT_FLAGS::CORJIT_FLAG_TIER0);
```

Comment:
If `QuickJit` is disabled, this may still generate instrumented tier 0 code. I figure this could happen with an `R2R -> OptimizationTierInstrumented` transition. If `QuickJit` is disabled, is that transition disallowed somehow?

Could add an assertion that `QuickJit` is enabled here to verify that this path is not reached when it is disabled.